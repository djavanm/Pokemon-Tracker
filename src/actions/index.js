export const setPokemon = pokemon => ({
  type: 'SET_POKEMON',
  pokemon
});

export const catchPokemon = pokemon => ({
  type: 'CATCH_POKEMON',
  pokemon
});

export const releasePokemon = id => ({
  type: 'RELEASE_POKEMON',
  id
});
