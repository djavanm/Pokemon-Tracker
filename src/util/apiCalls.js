export const getPokemon = async indentifier => {
  const url = `https://pokeapi.co/api/v2/pokemon/${indentifier}/`;
  const response = await fetch(url);
  if(!response.ok) {
    throw new Error('There was an error getting your pokemon.');
  };
  const pokemon = await response.json();
  return pokemon;
};
