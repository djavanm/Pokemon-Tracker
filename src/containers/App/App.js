import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { getPokemon } from '../../util/apiCalls';
import { setPokemon } from '../../actions';
import Nav from '../../components/Nav/Nav';
import SearchForm from '../SearchForm/SearchForm';
import PokemonCard from '../PokemonCard/PokemonCard';
import MyPC from '../MyPC/MyPC';
import MyTeam from '../MyTeam/MyTeam';
import { Route, Redirect } from 'react-router-dom';

export class App extends Component {
  async componentDidMount() {
    const { setPokemon } = this.props;
    try {
    const initialPokemon = await getPokemon('mew');
    setPokemon(initialPokemon);
    console.log(initialPokemon);
    } catch {
    console.log('error')
    }
  };

  render() {
    const  { currentPokemon } = this.props;
    return (
      <main className="App">
        <Nav />
        <Route  exact path='/' component={SearchForm} />
        <Route  exact path='/myPC' render={() => currentPokemon ? <MyPC /> : <Redirect to='/' /> } />
        <Route  exact path='/myTeam' render={() => currentPokemon ? <MyTeam /> : <Redirect to='/' /> }/>
        {currentPokemon && <PokemonCard /> }
      </main>
    )
  }
};

export const mapStateToProps = state => ({
  currentPokemon: state.currentPokemon
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ setPokemon }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(App);
