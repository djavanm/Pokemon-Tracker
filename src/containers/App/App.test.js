import React from 'react';
import ReactDOM from 'react-dom';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { shallow, mount } from 'enzyme';
import { setPokemon } from '../../actions';
import { getPokemon, getDescription } from '../../util/apiCalls';

jest.mock('../../util/apiCalls');

getDescription.mockImplementation(() => {
  return 'Bulbasaur is cool.';
});

getPokemon.mockImplementation(() => {
  return {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
  };
});

describe('App', () => {
  let wrapper;

  const mockPokemon = {
    name: 'Bulbasaur',
    description: 'Bulbasaur is cool.',
    gen: 'Generation 1',
    gif: 'https://projectpokemon.org/images/normal-sprite/bulbasaur.gif',
    height: 7,
    id: 1,
    image: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
    sprite: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png'
  };

  const mockCaughtPokemon = [mockPokemon];
  const mockTeamPokemon = [mockPokemon];
  const mockCurrentPokemon =  mockPokemon;
  const mockState = {
    currentPokemon: mockPokemon,
    caughtPokemon: mockCaughtPokemon,
    teamPokemon: mockTeamPokemon
  };

  beforeEach(() => {
  wrapper = shallow(<App />)
});

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      currentPokemon: mockPokemon,
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

  it('it calls dispatch with the setPokemon action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setPokemon(mockPokemon);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setPokemon(mockPokemon);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call getPokemon when mounting', () => {
  expect(getPokemon).toHaveBeenCalledWith('mew');
  });

});
