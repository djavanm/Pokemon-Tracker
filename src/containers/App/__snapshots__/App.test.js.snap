// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the data passed through 1`] = `
<main
  className="App"
>
  <Nav />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(SearchForm)",
        "type": [Function],
      }
    }
    exact={true}
    path="/"
  />
  <Route
    exact={true}
    path="/myPC"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/myTeam"
    render={[Function]}
  />
</main>
`;
